app/src/main/java/ru/woodymsk/socialapp/presentation/map_screen/MapScreenViewModel.kt

```diff
@@ -0,0 +1,297 @@
+import com.yandex.runtime.Error
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import ru.woodymsk.socialapp.domain.toBoundingBox
+import ru.woodymsk.socialapp.error.AppError.Companion.handleError
+import ru.woodymsk.socialapp.presentation.common.BaseViewModel
+import ru.woodymsk.socialapp.presentation.map_screen.model.MapEvents
+import ru.woodymsk.socialapp.presentation.map_screen.model.MapUiState
+import ru.woodymsk.socialapp.presentation.map_screen.model.SearchResponseItem
+import ru.woodymsk.socialapp.presentation.map_screen.model.SearchState
+import ru.woodymsk.socialapp.presentation.map_screen.model.SuggestHolderItem
+import ru.woodymsk.socialapp.presentation.map_screen.model.SuggestState
+import ru.woodymsk.socialapp.presentation.map_screen.model.toTextStatus
+import ru.woodymsk.socialapp.presentation.navigation.model.Screen
+import javax.inject.Inject
+
+class MapScreenViewModel @Inject constructor() : BaseViewModel() {
+
+    private val _uiState = MutableStateFlow(MapUiState())
+    val uiState: StateFlow<MapUiState> = _uiState.asStateFlow()
+
+    private var searchSession: Session? = null
+    private var currentRegion: VisibleRegion? = null
+    private var shouldShowSuggestions = true
+    private var zoomToSearchResult = false
+    private val searchManager =
+        SearchFactory.getInstance().createSearchManager(SearchManagerType.COMBINED)
+    private val suggestSession: SuggestSession = searchManager.createSuggestSession()
+    private val exceptionHandler = CoroutineExceptionHandler { _, exception ->
+        handleError(exception)
+    }
+
+    private val searchSessionListener = object : Session.SearchListener {
+        override fun onSearchResponse(response: Response) {
+            val items = response.collection.children.mapNotNull {
+                val point = it.obj?.geometry?.firstOrNull()?.point ?: return@mapNotNull null
+                SearchResponseItem(point, it.obj)
+            }
+            val boundingBox = response.metadata.boundingBox ?: return
+
+            _uiState.update {
+                it.copy(
+                    searchState = SearchState.Success(
+                        items = items,
+                        zoomToItems = zoomToSearchResult,
+                        itemsBoundingBox = boundingBox,
+                    ),
+                    suggestState = SuggestState.Off,
+                )
+            }
+        }
+
+        override fun onSearchError(error: Error) {
+            handleError(SearchState.Error.toTextStatus())
+        }
+    }
+
+    private val suggestSessionListener = object : SuggestSession.SuggestListener {
+        override fun onResponse(response: List<SuggestItem>) {
+            if (!shouldShowSuggestions) return
```

Comment:
```suggestion
            if (shouldShowSuggestions.not()) return
```

так более читаемо, мне кажется. Не очень люблю восклицательный знак)

Reply:
согласен